src/useGoogleAnalytics.ts

```diff
@@ -3,19 +3,31 @@ import GoogleAnalytics from 'react-ga4';
 import { useLocation } from 'react-router-dom';
 import { useOneTrust } from './useOneTrust';
 
-export const GoogleAnalyticsConsentValue = {
-    Denied: "denied",
-    Granted: "granted"
+export enum GoogleAnalyticsConsentValue {
+    Denied = "denied",
+    Granted = "granted"
 };
 
-export const useGoogleAnalytics = ({ googleAnalyticsId, debug, nonce, oneTrustScriptDomain }) => {
-    let location = useLocation();
-    const [initializeOneTrust] = useOneTrust({oneTrustScriptDomain, nonce});
+interface UseGoogleAnalyticsParams {
+    googleAnalyticsId?: string;
+    debug?: boolean;
+    nonce?: string;
+    oneTrustScriptDomain?: string;
+};
+
+export const useGoogleAnalytics = ({
+    googleAnalyticsId,
+    debug,
+    nonce,
+    oneTrustScriptDomain
+}: UseGoogleAnalyticsParams) => {
+    const location = useLocation();
+    const [initializeOneTrust] = useOneTrust({ oneTrustScriptDomain, nonce });
 
     const [initialized, setInitialized] = useState(false);
-    const [previousPage, setPreviousPage] = useState(null);
-    
-if (googleAnalyticsId && !initialized) {
+    const [previousPage, setPreviousPage] = useState<any>(null);
```

Comment:
don't we know the type that `previousPage` will hold? I am trying to understand why the type is `any`.

Reply:
I misunderstood the type, just changed it to string.